app/views/vendors/_vendorinput.html.erb

```diff
@@ -0,0 +1,17 @@
+<section>
+  <%= form_for ([@market, @vendor]) do |t| %>
+  <article class="form">
```

Comment:
Every HTML tag inside the form_for should be indented. Indentation may not seem important, but it greatly improves the readability of your code, which is huge when someone else comes to look at it (or you go to look at it six months down the road..)